.travis.yml

```diff
@@ -17,99 +17,69 @@ env:
+    - CONDA_DEPENDENCIES="${CONDA_MIN_DEPENDENCIES} seaborn clustalw=2.1 netcdf4 scikit-learn"
     - CONDA_CHANNELS='biobuilds conda-forge'
     - CONDA_CHANNEL_PRIORITY=True
     - NUMPY_VERSION=stable
+    - INSTALL_HOLE="true"
 
-
+  matrix:
+    # Run a coverage test
+    - COVERALLS="true" SETUP_CMD="${PYTEST_FLAGS} --cov=MDAnalysis"
+    - PYTHON_VERSION=3.5
+    - PYTHON_VERSION=3.4
+    - PYTHON_VERSION=2.7
+    - NUMPY_VERSION=1.10.4
+    - NUMPY_VERSION=dev  EVENT_TYPE="cron"
 
 matrix:
   fast_finish: true
   include:
-    - os : linux
-      env: NAME='minimal'
-           MEMLEAK='--with-memleak'
-           MAIN_CMD='pytest ${PYTEST_LIST} ${PYTEST_FLAGS}'
-
-    - os: linux
-      env: NAME="Doc"
+    - env: NAME="Doc"
            MAIN_CMD="cd package && python setup.py"
            SETUP_CMD="build_sphinx"
            BUILD_DOCS=true
            BUILD_CMD="cd ${TRAVIS_BUILD_DIR}/package && python setup.py build_ext --inplace"
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
+           INSTALL_HOLE="false"
 
-    - os: linux
-      env: NAME="Lint"
+    # todo update to default 3.6 (aka removing the line with PYTHON_VERSION here)
+    - env: NAME="Lint"
+           PYTHON_VERSION=2.7
            MAIN_CMD="pylint --rcfile=package/.pylintrc package/MDAnalysis && pylint --rcfile=package/.pylintrc testsuite/MDAnalysisTests"
-           SETUP_CMD=''
+           SETUP_CMD=""
            BUILD_CMD=""
-           CONDA_DEPENDENCIES="pylint backports.functools_lru_cache"
-           PIP_DEPENDENCIES=""
-
-    - os: linux
-      env: NAME='full'
-           NOSE_COVERAGE='--with-coverage --cover-package MDAnalysis'
-           PYTEST_COVERAGE='--cov=MDAnalysis'
-           SETUP_CMD="${PYTEST_FLAGS} ${PYTEST_COVERAGE}"
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
-           COVERALLS='true'
+           CONDA_DEPENDENCIES=""
+           INSTALL_HOLE="false"
 
     - os: osx
-      env: NAME='osx'
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
-
-    - os: linux
-      env: NAME='old numpy'
-           NUMPY_VERSION=1.10.4
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
+      env: PYTHON_VERSION=3.6
 
-    - os: linux
-      env: NAME='numpy dev'
-           NUMPY_VERSION=dev
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
-           EVENT_TYPE='cron'
-
-    - os: linux
-      env: NAME='full-py3.4'
-           PYTHON_VERSION=3.4
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
-
-    - os: linux
-      env: NAME='full-py3.5'
-           PYTHON_VERSION=3.5
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
-
-    - os: linux
-      env: NAME='full-py3.6'
-           PYTHON_VERSION=3.6
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
+    - os: osx
+      env: PYTHON_VERSION=2.7
 
+    - env: NAME='minimal'
+           CONDA_DEPENDENCIES=${CONDA_MIN_DEPENDENCIES}
+           MAIN_CMD="pytest testsuite/MDAnalysisTests/analysis"
```

Comment:
Also minimal now only runs on the analysis module. This should keep the time at a minimum